packages/extension-api/src/extension-api.d.ts

```diff
@@ -1659,4 +1659,159 @@ declare module '@podman-desktop/api' {
       options?: AuthenticationProviderOptions,
     ): Disposable;
   }
+
+  /**
+   * Namespace describing the environment Podman Desktop runs in.
+   */
+  export namespace env {
+    /**
+     * Indicates whether the users has telemetry enabled.
+     * Can be observed to determine if the extension should send telemetry.
+     */
+    export const isTelemetryEnabled: boolean;
+
+    /**
+     * An {@link Event} which fires when the user enabled or disables telemetry.
+     * `true` if the user has enabled telemetry or `false` if the user has disabled telemetry.
+     */
+    export const onDidChangeTelemetryEnabled: Event<boolean>;
+
+    /**
+     * Creates a new {@link TelemetryLogger telemetry logger}.
+     *
+     * @param sender The telemetry sender that is used by the telemetry logger.
+     * @param options Options for the telemetry logger.
+     * @returns A new telemetry logger
+     */
+    export function createTelemetryLogger(sender?: TelemetrySender, options?: TelemetryLoggerOptions): TelemetryLogger;
+  }
+
+  /**
+   * A special value wrapper denoting a value that is safe to not clean.
+   * This is to be used when you can guarantee no identifiable information is contained in the value and the cleaning is improperly redacting it.
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  export class TelemetryTrustedValue<T = any> {
+    readonly value: T;
+
+    constructor(value: T);
+  }
+
+  /**
+   * A telemetry logger which can be used by extensions to log usage and error telementry.
+   *
+   * A logger wraps around an {@link TelemetrySender sender} but it guarantees that
+   * - user settings to disable or tweak telemetry are respected, and that
+   * - potential sensitive data is removed
+   *
+   * It also enables an "echo UI" that prints whatever data is send and it allows the editor
+   * to forward unhandled errors to the respective extensions.
+   *
+   * To get an instance of a `TelemetryLogger`, use
+   * {@link env.createTelemetryLogger `createTelemetryLogger`}.
+   */
+  export interface TelemetryLogger {
+    /**
+     * An {@link Event} which fires when the enablement state of usage or error telemetry changes.
+     */
+    readonly onDidChangeEnableStates: Event<TelemetryLogger>;
+
+    /**
+     * Whether or not usage telemetry is enabled for this logger.
+     */
+    readonly isUsageEnabled: boolean;
```

Comment:
Instead of having `usageEnabled` and `errorEnabled` why didn't you use levels (log, warn, error, critical..)?

Reply:
e.g on vscode they have all, error, crash, off https://code.visualstudio.com/docs/getstarted/telemetry#_disable-telemetry-reporting

Reply:
@lstocchi https://github.com/microsoft/vscode/blob/main/src/vscode-dts/vscode.d.ts#L16821-L16829